### PR TITLE
Move to kubernetes ci 

### PR DESCRIPTION
**Pull request by clbx**
GMP 634

### CODE REVIEW

**Review by moosecanswim — DISMISSED** [review]
LGTM

**Review by tilleryd — CHANGES_REQUESTED** [review]
This is a component library installed via NPM. We bundle the assets and publish them to the public NPM repo, see https://www.npmjs.com/package/gm-ui-components. We don't need any assets built and publish to Artifactory.

**Comment by clbx** [comment]
@tilleryd this doesn't publish any assets to artifactory

File: .buildkite/init-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+agents:
+  queue: k8s-agent
+
+steps:
+  # Build the build image will will do that if there is an update to the dockerfile or pipeline
```
Comment (tilleryd) [inline .buildkite/init-pipeline.yaml:5]: @clbx Can you re-word this? It's a bit confusing.

File: .buildkite/workload-pipelines/build-image.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+steps:
+  - key: build-base-image
+    name: ":docker: Build Control Build Image"
```
Comment (tilleryd) [inline .buildkite/workload-pipelines/build-image.yaml:3]:
```suggestion
    name: ":docker: Build gm-ui-components Image"
```

File: .buildkite/dockerfiles/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+# Add Node 18 repository
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash
+
+RUN apt update -y
+
+RUN apt install -y gcc g++ make nodejs 
+
+
+
```
Comment (tilleryd) [inline .buildkite/dockerfiles/Dockerfile:15]:
```suggestion
```

File: .buildkite/init-pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                command: buildkite-agent pipeline upload .buildkite/workload-pipelines/build-image.yaml
+                label: "Upload build-image pipeline"
+  - wait
+
+  - label: ":pipeline: default build pipeline config"
+    command: buildkite-agent pipeline upload .buildkite/workload-pipelines/pipeline.yaml
+    if: build.source != "schedule" && build.tag == null
+    
+  
+
```
Comment (tilleryd) [inline .buildkite/init-pipeline.yaml:24]:
```suggestion
```

**Comment by tilleryd** [comment]
@joshuaswickirl Derp, I see that now.

File: .buildkite/workload-pipelines/build-image.yaml (hunk 2 above)
Comment (clbx) [reply .buildkite/workload-pipelines/build-image.yaml]: Good catch

**Review by tilleryd — CHANGES_REQUESTED** [review]
I recant my previous PR rejection. I just have a handful of small improvements requested now.

File: .buildkite/init-pipeline.yaml (hunk 1 above)
Comment (clbx) [reply .buildkite/init-pipeline.yaml:5]: Done

Comment (tilleryd) [reply .buildkite/init-pipeline.yaml:5]: Hmm, still confusing.

File: .buildkite/init-pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+agents:
+  queue: k8s-agent
+
+steps:
+  # Build the build image will will do that if there is an update to the dockerfile or pipeline
```
Comment (tilleryd) [inline .buildkite/init-pipeline.yaml:5]:
```suggestion
  # If there's an update to the Dockerfile or pipeline, build the image.
```

**Review by tilleryd — APPROVED** [review]

**Pull request merged**